src/components/Badge/Badge.tsx

```diff
@@ -23,6 +23,8 @@ export interface BadgeProps {
    * @default 'medium'
    */
   size?: Size;
+  /** Pass a custom accessibilityLabel (overrides status and progress labels) */
+  customAccessibilityLabel?: string;
```

Comment:
This PR looks good to me but I worry that this prop might be misunderstood and get misused. I like that it's different from the default `accessibilityLabel` prop but I'm not sure the name tells you enough about what it does. 

What about something terribly verbose?
`statusAndProgressLabelOverride`